File: 2020/day_01/day_01.py
with open('in') as f:
    data = list(map(int, f.readlines()))


def part_one():
    for i in range(len(data)):
        for j in range(i, len(data)):
            if data[i] + data[j] == 2020:
                return data[i] * data[j]


def part_two():
    for i in range(len(data)):
        for j in range(i, len(data)):
            for k in range(j, len(data)):
                if data[i] + data[j] + data[k] == 2020:
                    return data[i] * data[j] * data[k]


if __name__ == '__main__':
    print(f'Part One: {part_one()}')
    print(f'Part Two: {part_two()}')

File: 2020/day_01/day_01.rs
use std::fs;

fn main() {
    let f = fs::read_to_string("in").unwrap();
    let data: Vec<i32> = f.lines().map(|l| l.trim().parse().unwrap()).collect();

    println!("Part One: {}", part_one(&data));
    println!("Part Two: {}", part_two(&data));
}

fn part_one(data: &Vec<i32>) -> i32 {
    for a in data {
        if let Some(res) = data.iter().find(|x| *a + *x == 2020) {
            return *a * *res;
        }
    }

    0
}

fn part_two(data: &Vec<i32>) -> i32 {
    for a in data {
        for b in data {
            for c in data {
                if *a + *b + *c == 2020 {
                    return *a * *b * *c;
                }
            }
        }
    }

    0
}